### PR TITLE
firecracker: 1.9.0 -> 1.9.1, build from source

### PR DESCRIPTION
**Pull request by usertam**
Build from source instead of using prebuilt binaries.
- Work around a dependency compilation error. 
- Skip tests that will fail in sandbox.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by usertam** [comment]
cc @thoughtpolice @qjoly @techknowlogick

File: pkgs/by-name/fi/firecracker/package.nix [hunk 1]
```diff
@@ -1,59 +1,63 @@
...
+  ];
 
   installPhase = ''
     mkdir -p $out/bin
-    install -D firecracker $out/bin/firecracker
-    install -D jailer      $out/bin/jailer
+    releaseDir="build/cargo_target/${stdenv.hostPlatform.rust.rustcTarget}/release"
+    for bin in $(find $releaseDir -maxdepth 1 -type f -executable); do
+      install -Dm555 -t $out/bin $bin
+    done
```
Comment (Aleksanaa) [inline pkgs/by-name/fi/firecracker/package.nix:60]: runHook preInstall, runHook postInstall

**Review by Aleksanaa — APPROVED** [review]
Please reformat `packages.nix`. Others seems fine, please test functionality carefully to avoid regression.

**Review by techknowlogick — APPROVED** [review]

**Review by thoughtpolice — APPROVED** [review]
Impressive, I suspect there have been a lot of fixes upstream to make it easier these days. It overall looks OK to me. But we'll have to see how this shakes out; it's possible the maintenance burden will be greater, and this approach will be more fragile &mdash; but maybe it will remain relatively stable too; I've seen it go both ways.

A future update I'd like to see with expressions like this is an update script to try and handle all the tedious logic like `outputHashes` handling, and so that the Ryan bot can do work for us. But I'm not going to block the PR on that.

**Comment by usertam** [comment]
I have tested `firecracker` with `/dev/kvm`, and it works with the example in [getting-started](https://github.com/firecracker-microvm/firecracker/blob/main/docs/getting-started.md). I will test against `jailer` and others when I have time.

I did some work on `passthru.updateScript` previously on another package so I think I can do that in the future.

**Review by techknowlogick — APPROVED** [review]

**Review by Scrumplex — APPROVED** [review]

**Pull request merged**